### DISCUSSION TITLE
Workflow limit clasification

### DISCUSSION THREAD

**Discussion by bxff** [community/community#26481] [Actions]
<p>Ok, I just started using Github actions and am I little confused over some of the limits given on the page: <a href="https://help.github.com/en/actions/getting-started-with-github-actions/about-github-actions#usage-limits" rel="nofollow noopener">https://help.github.com/en/actions/getting-started-with-github-actions/about-github-actions#usage-limits</a></p>
<p>I get the first on that a job as to finish in 6 hours or it would be failed but others are a little confusing,</p>
<p><code>**Workflow run time**</code> - What does that mean, that a workflow has a limit of 72 hours, but what defines a workflow? The x.yml file in the .github/workflow directory or the one repos workflow all in one? And if its the x.yml then cant some one just create more of them to get around the problem?</p>
<p>And one last thing, does the time reset? I am asumming it does ever mounth as the plane section tells the actions limit for private repos but I cant tell in the documentation.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/bin0x00">@bin0x00</a> ,</p>
<p>The <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/configuring-a-workflow#about-workflows" rel="nofollow noopener">workflow</a> is defined in yaml file under ‘.github/workflows’ directory in the root of your repository.</p>
<p>The ‘worklfow run time’ limit is applied to echo workflow in your repo, not the one repos workflow all in one.</p>
<p>When the run time limit is reached, the workflow will be failed.</p>
<p>If the workflow contains mutliple jobs/steps, it’s possible to move some of them to a new workflow, and get rid of the limit, but since echo workflow are individual so data sharing/transfer will be a problem. (Possible using sharing file server, selfthosted runner, artifacts…etc)</p>
<p>There is no time reset for the limit, when you start a new workflow or re-run the failed workflow, it will obey this rule.</p>
<p>Hope it’s clear and helpful.</p>
